Dockerfile

```diff
@@ -30,23 +30,40 @@ RUN dd bs=1024 count=2880 if=/dev/zero of=esp.img && \
 
 FROM docker.io/centos:centos8
 
-RUN dnf install -y python3 python3-requests && \
+RUN dnf install -y gcc python3 python3-requests python3-devel && \
     curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/tripleo_repos/main.py | python3 - -b master current && \
     dnf update -y && \
-    dnf install -y python3-gunicorn openstack-ironic-api openstack-ironic-conductor crudini \
+    dnf install -y python3-gunicorn crudini OpenIPMI ipmitool \
         iproute dnsmasq httpd qemu-img iscsi-initiator-utils parted gdisk psmisc \
-        mariadb-server genisoimage python3-ironic-prometheus-exporter \
+        mariadb-server genisoimage \
         python3-jinja2 python3-sushy-oem-idrac && \
     dnf clean all && \
     rm -rf /var/cache/{yum,dnf}/*
 
+RUN /usr/bin/pip3 install pymysql
+RUN /usr/bin/pip3 install git+https://opendev.org/openstack/ironic.git@master
+
+RUN /usr/bin/pip3 install ironic-prometheus-exporter
```

Comment:
Please no pip install by default. Pip install is a terrible idea most of the time, this pulls in an absolutely untested combination of source packages and their dependencies.